partcad/src/partcad/part_factory_enrich.py

```
@@ -13,76 +13,73 @@
+
+            if ";" in source_part_name:
+                self.source_part_name = source_part_name.split(";")[0]
+                suffix = source_part_name.split(";")[1]
+                self.extra_with = [p.split("=") for p in suffix.split(",")]
+            else:
+                self.source_part_name = source_part_name
+                self.extra_with = []
+
+            self._create(config)
+
+    async def instantiate(self, part):
```

Ruff (0.8.2)

`instantiate` is too complex (17 > 10)

(C901)